lib/src/features/user/ui/perf_stats_screen.dart

```diff
@@ -0,0 +1,457 @@
+import 'package:flutter/cupertino.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_riverpod/flutter_riverpod.dart';
+import 'package:intl/intl.dart';
+
+import 'package:lichess_mobile/src/constants.dart';
+import 'package:lichess_mobile/src/features/auth/data/auth_repository.dart';
+import 'package:lichess_mobile/src/utils/duration.dart';
+import 'package:lichess_mobile/src/utils/style.dart';
+import 'package:lichess_mobile/src/common/lichess_colors.dart';
+import 'package:lichess_mobile/src/common/lichess_icons.dart';
+import 'package:lichess_mobile/src/common/models.dart';
+import 'package:lichess_mobile/src/features/user/data/user_repository.dart';
+import 'package:lichess_mobile/src/features/user/model/user.dart';
+import 'package:lichess_mobile/src/utils/l10n_context.dart';
+import 'package:lichess_mobile/src/widgets/feedback.dart';
+import 'package:lichess_mobile/src/widgets/platform.dart';
+
+final perfStatsProvider = FutureProvider.autoDispose
+    .family<UserPerfStats, UserPerfStatsParameters>((ref, perfParams) async {
+  final userRepo = ref.watch(userRepositoryProvider);
+  final either = await userRepo
+      .getUserPerfStatsTask(
+        perfParams.username,
+        perfParams.perf,
+      )
+      .run();
+
+  return either.match((error) => throw error, (data) {
+    ref.keepAlive();
+    return data;
+  });
+});
+
+const _customOpacity = 0.6;
+const _defaultStatFontSize = 12.0;
+const _defaultValueFontSize = 18.0;
+const _titleFontSize = 18.0;
+
+class PerfStatsScreen extends ConsumerWidget {
+  const PerfStatsScreen(
+      {required this.username, required this.perf, super.key});
+
+  final String username;
+  final Perf perf;
+
+  @override
+  Widget build(BuildContext context, WidgetRef ref) {
+    return ConsumerPlatformWidget(
+        ref: ref, androidBuilder: _androidBuilder, iosBuilder: _iosBuilder);
+  }
+
+  Widget _androidBuilder(BuildContext context, WidgetRef ref) {
+    return Scaffold(
+        appBar: AppBar(
+          titleSpacing: 0,
+          title: Row(
+            mainAxisAlignment: MainAxisAlignment.start,
+            children: [
+              Icon(perf.icon),
+              const SizedBox(width: 5),
+              Expanded(
+                child: FittedBox(
```

Comment:
This `FittedBox` allows the text to be scaled down in case it is too large to fit in one line, instead of the word placed in two lines (thus keeping the UI more natural, in my opinion).

Reply:
ok thanks! I've never used that widget, thus the asking